pkg/limits/ingest_limits.go

```diff
@@ -0,0 +1,669 @@
+package limits
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/ring"
+	"github.com/grafana/dskit/services"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/twmb/franz-go/pkg/kgo"
+
+	"github.com/grafana/loki/v3/pkg/kafka"
+	"github.com/grafana/loki/v3/pkg/kafka/client"
+	"github.com/grafana/loki/v3/pkg/logproto"
+	"github.com/grafana/loki/v3/pkg/util"
+	"github.com/grafana/loki/v3/pkg/util/constants"
+	util_log "github.com/grafana/loki/v3/pkg/util/log"
+)
+
+const (
+	// Ring
+	RingKey  = "ingest-limits"
+	RingName = "ingest-limits"
+
+	// Kafka
+	consumerGroup = "ingest-limits"
+)
+
+var (
+	tenantPartitionDesc = prometheus.NewDesc(
+		constants.Loki+"_ingest_limits_partitions",
```

Comment:
I believe this is no longer recommended, almost all of our new code adds `loki` as a string literal.